src/roiextractors/extractors/hdf5imagingextractor/hdf5imagingextractor.py

```diff
@@ -71,13 +74,16 @@ def __init__(
         (
             self._num_channels,
             self._num_frames,
-            self._size_x,
-            self._size_y,
+            self._rows,
+            self._columns,
         ) = get_video_shape(self._video)
 
-        if len(self._video.shape) == 3:
-            # check if this converts to np.ndarray
-            self._video = self._video[np.newaxis, :]
+        # I am assuming that the videos come here with self._num_channels in the first axis
+        self._data_has_channels_axis = self._video.ndim > 3
+        if self._data_has_channels_axis:
+            self._standard_video = self._video.lazy_transpose([1, 2, 3, 0])
```

Comment:
This sort of map could be nice, but I would avoid having both `_video` and `_standard_video`. Perhaps define a strictly internal `initial_video` then define `self._video = initial_video[proper_axis_order]`?

Reply:
> You want different labels/variable names for them or am I missing something?

I don't see a point to having both views as class attributes, better to just have one attribute that is the expected shape (the `self._standard_video` here, but named `self._video` since no need to adjust the name to differentiate).

Reply:
All right, I get you and yes, I think that's better. Just having an internal variable that is already on the standard shape is better. Not point in keeping a point to the original form.

Reply:
Done.